esp-hal/src/uart.rs

```diff
@@ -1719,6 +1782,16 @@ mod asynch {
                 }
             }
         }
+
+        /// Sets the buffer to be used for async reads.
+        pub fn set_buffer<const N: usize>(&mut self, buffer: &'static mut heapless::Deque<u8, N>)
```

Comment:
I don't really managing global state like this tbh, I think this should be managed by the user.